src/applications/static-pages/subscribeComponentAnalyticsEvents.js

```diff
@@ -0,0 +1,50 @@
+import recordEvent from 'platform/monitoring/record-event';
+
+const analyticsEvents = [
+  {
+    componentName: 'Modal',
+    prefix: 'modal',
+    actions: [{ action: 'show', name: 'int-modal-click' }],
+  },
+  {
+    componentName: 'AdditionalInfo',
+    prefix: 'additional-info',
+    actions: [
+      { action: 'expand', name: 'int-additional-info-expand' },
+      { action: 'collapse', name: 'int-additional-info-collapse' },
+    ],
+  },
+];
+
+export default function subscribeComponentAnalyticsEvents() {
+  document.body.addEventListener('component-library-analytics', e => {
+    // Is it a component we are tracking?
+    const component = analyticsEvents.find(
+      c => c.componentName === e.detail.componentName,
+    );
+
+    // Is it an action we are tracking?
+    if (component && component.actions) {
```

Comment:
You can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here:

```suggestion
    if (component?.actions) {
```